apps/website/src/pages/blog/index.tsx

```diff
@@ -1,64 +1,280 @@
-import { Button, Shadow, Tag, Text } from '@status-im/components'
+import { Avatar, Button, Shadow, Tag, Text } from '@status-im/components'
+import { useInfiniteQuery } from '@tanstack/react-query'
+import { useRouter } from 'next/router'
 
+import { formatDate } from '@/components/chart/utils/format-time'
+import { Link } from '@/components/link'
 import { AppLayout } from '@/layouts/app-layout'
+import { getPosts, getTags } from '@/lib/ghost'
 
-import type { Page } from 'next'
+import type { PostOrPage, PostsOrPages, Tags } from '@tryghost/content-api'
+import type { GetStaticProps, InferGetStaticPropsType, Page } from 'next'
+
+export const getStaticProps: GetStaticProps<{
+  tags: Tags
+  posts: PostOrPage[]
+  meta: PostsOrPages['meta']
+}> = async () => {
+  const { posts, meta } = await getPosts()
+  const tags = await getTags()
+
+  return {
+    props: {
+      tags,
+      posts,
+      meta,
+    },
+  }
+}
+
+type Props = InferGetStaticPropsType<typeof getStaticProps>
+
+const BlogPage: Page<Props> = props => {
+  const { tags: initialTags, posts, meta } = props
+
+  const tags = new Map<string, { name?: string; slug: string }>(
+    initialTags.map(tag => [tag.slug, tag])
+  )
+
+  const router = useRouter()
+  const { tag } = router.query as { tag?: string }
+
+  const {
+    data,
+    // error,
+    fetchNextPage,
+    hasNextPage,
+    // isFetching,
+    // isFetchingNextPage,
+    // status,
+    // isFetched,
+  } = useInfiniteQuery({
+    queryKey: ['posts', tag],
+    queryFn: async ({ pageParam: page, queryKey }) => {
+      const [, tag] = queryKey
+
+      const { posts, meta } = await getPosts({ page, tag })
+
+      let filteredPosts = posts
+      if (tag && tags.has(tag)) {
+        filteredPosts = filteredPosts.filter(
+          post => post.primary_tag?.slug === tag
+        )
+      }
```

Comment:
🤔

Reply:
Removed filters for now, will tackle as a follow-up.